### ISSUE TITLE
Always serve minified assets regardless of exclusion rules?

### ISSUE THREAD

**Issue by lkraav** [wpsharks/comet-cache#506]
For example, WC checkout pages have DONOTCACHEPAGE constant set. But this shouldn't affect asset minification. Thoughts?

**Comment by jaswrks** [comment]
I tend to agree. However, this seems to create some confusion (such as [what @raamdev noted here](https://github.com/websharks/zencache/issues/486#issuecomment-106561065)). Definitely worth looking at IMO though. I think we just need to work on clarifying this for site owners.

**Comment by jaswrks** [comment]
@raamdev What are your thoughts on this?

**Comment by raamdev** [comment]
@jaswsinc Allowing minificaiton when caching is disabled will definitely introduce some confusion. Can the HTML Compressor be used when ZenCache is disabled (i.e., caching is turned off)? At present, no. The HTML Compressor is a feature of ZenCache and when ZenCache (or ZenCache caching) is disabled, so is the HTML Compressor.

Making it possible for the HTML Compressor to run when caching is disabled (either via `DONOTCACHEPAGE` or by disabling caching in **ZenCache → Plugin Options → Enable/Disable**) would, in my mind, be separating the HTML Compressor from ZenCache. 

It seems that it would make more sense, in that case, to offer the HTML Compressor as a separate plugin (in addition to being part of ZenCache Pro). Then, if someone wants to run the HTML Compressor separately from ZenCache (i.e., to have everything minified even if `DONOTCACHEPAGE` is set or if ZenCache is disabled), then the site owner could install and use the independent plugin version of the HTML Compressor. That independent plugin could be a free download for any ZenCache Pro users, and perhaps available for purchase for non-ZenCache Pro users.

---

The only other option, in my mind, would be to add a filter to ZenCache Pro that would allow a site owner to override the default behavior of disabling HTML Compression when `DONOTCACHEPAGE` is set to true. I'd be fine with a filter, or with working on creating a distributable plugin version of the HTML Compressor.

**Comment by KTS915** [comment]
In my view, it would be great to have the HTML Compressor working even if caching is disabled. Whether you provide that capability via an independent plugin or a filter doesn't matter to me.

**Comment by jaswrks** [comment]
:+1:

**Comment by clavaque** [comment]
This has my vote. I think the filter option would be good enough.

I believe that even when caching is disabled for the page, it should be possible to have the combined/minified resources (i.e. scripts, styles) cached. Caching these could be a setting in the HTML Compression panel. Otherwise, minifying/combining the resources in real-time would add time and use more resources.

I see that even in a page with caching disabled (e.g. it has an s2 pro-form), the static files are served from the CDN when that is enabled. Having the CDN benefit makes sense and doesn't affect the page's cache being disabled. And neither does the caching of minified/combined resources, in my opinion.

**Comment by jaswrks** [comment]
Noting that this issue now has a low priority in my mind, because of HTTP/2. So while I still agree that a new plugin for the HTML Compressor and/or a filter would be nice, I think our efforts should be focused on HTTP/2 instead of on the HTML Compressor.

With HTTP/2, most (all?) of what the HTML Compressor does is no longer needed whatsoever. And, in fact, it would be discouraged if your server supports HTTP/2. https://http2.github.io/
